sootty/save.py

```diff
@@ -1,27 +1,96 @@
+            pass
+        else:
+            with open(savefile, "w") as stream:
+                if len(lines) >= 500:
+                    print(
+                        "Saved query limit reached. Deleting least recent query to accommodate new query.",
+                        file=sys.stderr,
+                    )
+                    stream.truncate(0)
+                    for key in lines:
+                        stat = lines.pop(key)
+                        break
+                    if stat is None:  # No lines to delete/Error
+                        raise SoottyError("Error deleting least recent query.")
+                    yaml.dump(lines, stream, sort_keys=False, width=float("inf"))
+
+        with open(savefile, "a+") as stream:
+            query_write(
+                savefile, stream, save, name, wires, br, length, start, end, display
+            )
+
+    else:
+        # Creating new savefile as no savefiles found for sootty
+        with open(savefile, "w") as stream:
+            query_write(
+                savefile, stream, save, name, wires, br, length, start, end, display
+            )
+
+
+def query_write(
+    savefile_path, savefile, save, name, wires, br, length, start, end, display
+):
+    with open(savefile_path, "r+") as stream:
+        lines = yaml.safe_load(stream)
+        if save in lines:
+            del lines[save]  # Deleting outdated query
+            overwrite_dict = {
+                save: {
+                    "query": query_build(name, wires, br, length, start, end, display),
+                    "date": str(datetime.datetime.now()),
+                }
+            }
+            lines.update(
+                overwrite_dict
+            )  # Essentially replacing the old query with the new dict
+            stream.truncate(0)
+            yaml.dump(
+                lines, savefile, sort_keys=False, width=float("inf")
+            )  # Dumping the overwritten query to the file, forcing no inline output
+        else:
+            savefile.write(save + ":\n")
+            savefile.write("  query:")
+            savefile.write(query_build(name, wires, br, length, start, end, display))
+            savefile.write("\n")
+            savefile.write("  date: " + str(datetime.datetime.now()) + "\n")
+
+
+def query_build(name, wires, br, length, start, end, display):
     """
     Constructing the query using conditionals
     """
-    with open(save, "w") as wf:
-        wf.truncate(0)
-        if name:
-            wf.write(' -f "' + name + '"')
-        if wires:
-            wf.write(' -w "' + wires + '"')
-        if br:
-            wf.write(' -b "' + br + '"')
-        if length:
-            wf.write(' -l "' + str(length) + '"')
-        if start:
-            wf.write(' -s "' + start + '"')
-        if end:
-            wf.write(' -e "' + end + '"')
-        if display:
-            wf.write(" -d")
+    cmd = ""
+    if name:
+        cmd += ' -f "' + name + '"'
```

Comment:
Python string formatting would be helpful here, since it automatically converts the variable to the correct type:

```f' -f "{name}"'```